e2e/__tests__/cats.test.js

```diff
@@ -0,0 +1,226 @@
+const request = require('../request');
+const db = require('../db');
+const User = require('../../lib/models/user');
+const { signupUser } = require('../data-helpers');
+
+describe('Cats API', () => {
+  beforeEach(() => db.dropCollection('users'));
+  beforeEach(() => db.dropCollection('cats'));
+
+  const normalUser = {
+    email: 'normal@normal.com',
+    password: 'abc123'
+  };
+
+  const adminUser = {
+    email: 'admin@admin.com',
+    password: 'abc123'
+  };
+
+  const cat1 = {
+    name: 'CMO',
+    breed: 'American Shorthair',
+    toes: 5,
+    colors: 'Orange & White',
+    claws: true,
+    whiskers: 'Long'
+  };
+
+  const cat2 = {
+    name: 'Ender',
+    breed: 'Mumbai',
+    toes: 5,
+    colors: 'Black with Red Undertones',
+    claws: true,
+    whiskers: 'Normal'
+  };
+
+  function signinAdminUser(admin = adminUser) {
+    return request
+      .post('/api/auth/signin')
+      .send(admin)
+      .expect(200)
+      .then(({ body }) => body);
+  }
+
+  it('posts a cat with user permission', () => {
+    return signupUser(adminUser)
+      .then(user => {
+        return User.updateById(user._id, {
+          $addToSet: {
+            roles:'admin'
+          }
+        });
+      })
+      .then(() => {
+        return Promise.all([
```

Comment:
you only need a promise.all when you are running more than one function which you need return values from. Here you could just do ```return signinAdminUser()```
```.then(adminUser => {```

etc etc